olive/resource_path.py

```diff
@@ -490,8 +535,25 @@ def save_to_dir(self, dir_path: Union[Path, str], name: str = None, overwrite: b
             )
             new_path.parent.mkdir(parents=True, exist_ok=True)
             shutil.move(temp_dir / "named-outputs" / self.config.output_name / self.config.relative_path, new_path)
-
+            self.write_flag_file(dir_path, new_path)
         return str(new_path)
 
 
-OLIVE_RESOURCE_ANNOTATIONS = Optional[Union[Path, str, ResourcePath, ResourcePathConfig]]
+OLIVE_RESOURCE_ANNOTATIONS = Optional[Union[str, Path, ResourcePath, ResourcePathConfig]]
+OLIVE_DOWNLOAD_SUCCESSFUL_FLAG = ".olive_resource_flag"
+LOCAL_CACHE_DIR = "./cache/aml_datastore_data"
+
+
+def get_local_path(resource_path: OLIVE_RESOURCE_ANNOTATIONS) -> str:
```

Comment:
Instead of doing all of the flag manipulation, I think it might be easier to do cache the remote resource like in cache https://github.com/microsoft/Olive/blob/main/olive/cache.py#L113

Reply:
Coool, updated.